airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -231,10 +231,29 @@ private static AirbyteStream removeIncrementalWithoutPk(final AirbyteStream stre
   public Set<JdbcPrivilegeDto> getPrivilegesTableForCurrentUser(final JdbcDatabase database, final String schema) throws SQLException {
     return database.query(connection -> {
       final PreparedStatement ps = connection.prepareStatement(
+          // c.relkind = 'm' means Materialized View
+          // c.relacl is null or s[2] = 'r' means either null (all grants) or "r" - grants for read
+          // coalesce ('SELECT') as privilege_type\n" is set as it to match Union query, verification is done
+          // is "where" clause
           "SELECT DISTINCT table_catalog, table_schema, table_name, privilege_type\n"
               + "FROM   information_schema.table_privileges\n"
-              + "WHERE  grantee = ? AND privilege_type = 'SELECT'");
-      ps.setString(1, database.getDatabaseConfig().get("username").asText());
+              + "WHERE  grantee = ? AND privilege_type = 'SELECT'"
+              + "union all\n"
+              + "select \n"
+              + "\tr.rolname as table_catalog,\n"
+              + "\tn.nspname as table_schema,\n"
+              + "\tc.relname as table_name,\n"
+              + "    coalesce ('SELECT') as privilege_type\n"
+              + "from \n"
+              + "    pg_class c\n"
+              + "    join pg_namespace n on n.oid = relnamespace\n"
+              + "    join pg_roles r on r.oid = relowner,\n"
+              + "    unnest(coalesce(relacl::text[], format('{%s=arwdDxt/%s}', rolname, rolname)::text[])) acl, \n"
+              + "    regexp_split_to_array(acl, '=|/') s\n"
+              + "   where r.rolname = ? and  nspname = 'public' and c.relkind = 'm' and (c.relacl is null or s[2] = 'r');");
```

Comment:
[OPTIONAL] As we migrate to Java 17, could we use Text Blocks https://blogs.oracle.com/javamagazine/post/text-blocks-come-to-java here? I believe it will help to make code changes faster in future.

Reply:
done, thanks